common/src/main/java/com/gregtechceu/gtceu/common/data/GTItems.java

```diff
@@ -71,38 +71,54 @@ public class GTItems {
     //////////////////////////////////////
     //*****     Material Items    ******//
     //////////////////////////////////////
+
+
+
+
+    // Externally Accessible Values
     public static Table<TagPrefix, Material, ItemEntry<TagPrefixItem>> MATERIAL_ITEMS;
+    public final static Table<MaterialToolTier, GTToolType, ItemEntry<GTToolItem>> TOOL_ITEMS =
+            ArrayTable.create(GTRegistries.MATERIALS.values().stream().filter(mat -> mat.hasProperty(PropertyKey.TOOL)).map(Material::getToolTier).toList(),
+                    Arrays.stream(GTToolType.values()).toList());
 
-    public static void generateMaterialItems() {
+    // Table Builders
+    private static final ImmutableTable.Builder<TagPrefix, Material, ItemEntry<TagPrefixItem>> MATERIAL_ITEMS_BUILDER = ImmutableTable.builder();
+
+    // Material Item Generation
+    private static boolean allowedItemMaterial(TagPrefix tagPrefix, Material material) {
+        boolean doGenerateItem = tagPrefix.doGenerateItem(material);
+        return doGenerateItem;
```

Comment:
why not just `return tagPrefix.doGenerateItem()` ?
actually, why have this method at all?

Reply:
I could see it being enough to just return the tagPrefix, however I wanted to make it explicit enough to make sure it's clear that it's the rule for if the item is being generated.
There are other places in GTBlocks that do that for readability.